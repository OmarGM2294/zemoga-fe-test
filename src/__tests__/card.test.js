import React from 'react'

import { create } from 'react-test-renderer'

import Card from '../components/card'

const data = {
  name: 'Kanye West',
  description:
    'Born in Atlanta and raised in Chicago, West was first known as a producer for Roc-A-Fella Records in the early 2000s, producing singles for several mainstream artists.',
  category: 'entertainment',
  picture: 'kanye@2x.png',
  lastUpdated: '2020-03-10T23:08:57.892Z',
  votes: {
    positive: 23,
    negative: 36,
  },
}

describe('Card', () => {
  it('Should match snapshot (it renders correctly)', () => {
    const component = create(<Card data={data} type={'List'} />)

    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
