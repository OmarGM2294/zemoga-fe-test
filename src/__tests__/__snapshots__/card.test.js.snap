// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Should match snapshot (it renders correctly) 1`] = `
<div
  className="undefined undefined"
  style={
    Object {
      "backgroundImage": "url('/images/cards/kanye@2x.png')",
    }
  }
>
  <div>
    <div>
      <div>
        <h3>
          Kanye West
        </h3>
        <p>
          Born in Atlanta and raised in Chicago, West was first known as a producer for Roc-A-Fella Records in the early 2000s, producing singles for several mainstream artists.
        </p>
      </div>
      <div>
        <p>
          1 year
           ago in 
          <span>
            entertainment
          </span>
        </p>
        <div>
          <button
            aria-label="Vote positive"
          >
            <div>
              <div
                style={
                  Object {
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "left": 0,
                    "margin": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <img
                  alt="Vote positive"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style={
                    Object {
                      "border": "none",
                      "bottom": 0,
                      "boxSizing": "border-box",
                      "display": "block",
                      "height": 0,
                      "left": 0,
                      "margin": "auto",
                      "maxHeight": "100%",
                      "maxWidth": "100%",
                      "minHeight": "100%",
                      "minWidth": "100%",
                      "objectFit": undefined,
                      "objectPosition": undefined,
                      "padding": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "visibility": "hidden",
                      "width": 0,
                    }
                  }
                />
              </div>
            </div>
          </button>
          <button
            aria-label="Vote negative"
          >
            <div>
              <div
                style={
                  Object {
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "left": 0,
                    "margin": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <img
                  alt="Vote negative"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style={
                    Object {
                      "border": "none",
                      "bottom": 0,
                      "boxSizing": "border-box",
                      "display": "block",
                      "height": 0,
                      "left": 0,
                      "margin": "auto",
                      "maxHeight": "100%",
                      "maxWidth": "100%",
                      "minHeight": "100%",
                      "minWidth": "100%",
                      "objectFit": undefined,
                      "objectPosition": undefined,
                      "padding": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "visibility": "hidden",
                      "width": 0,
                    }
                  }
                />
              </div>
            </div>
          </button>
          <button
            aria-label="Vote now"
          >
            Vote Now
          </button>
        </div>
      </div>
    </div>
    <div>
      <div
        style={
          Object {
            "flexBasis": "38.98%",
          }
        }
      >
        <div>
          <div>
            <div
              style={
                Object {
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "left": 0,
                  "margin": 0,
                  "overflow": "hidden",
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <img
                alt="Vote positive"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style={
                  Object {
                    "border": "none",
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": 0,
                    "left": 0,
                    "margin": "auto",
                    "maxHeight": "100%",
                    "maxWidth": "100%",
                    "minHeight": "100%",
                    "minWidth": "100%",
                    "objectFit": undefined,
                    "objectPosition": undefined,
                    "padding": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "visibility": "hidden",
                    "width": 0,
                  }
                }
              />
            </div>
          </div>
          <span>
            38.98%
          </span>
        </div>
      </div>
      <div
        style={
          Object {
            "flexBasis": "61.02%",
          }
        }
      >
        <div>
          <div>
            <div
              style={
                Object {
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "left": 0,
                  "margin": 0,
                  "overflow": "hidden",
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <img
                alt="Vote negative"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style={
                  Object {
                    "border": "none",
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": 0,
                    "left": 0,
                    "margin": "auto",
                    "maxHeight": "100%",
                    "maxWidth": "100%",
                    "minHeight": "100%",
                    "minWidth": "100%",
                    "objectFit": undefined,
                    "objectPosition": undefined,
                    "padding": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "visibility": "hidden",
                    "width": 0,
                  }
                }
              />
            </div>
          </div>
          <span>
            61.02%
          </span>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div>
      <div>
        <div>
          <div
            style={
              Object {
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "left": 0,
                "margin": 0,
                "overflow": "hidden",
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <img
              alt="Most negative votes"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": undefined,
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "visibility": "hidden",
                  "width": 0,
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
